test/e2e/main_test.go

```diff
@@ -3,61 +3,166 @@ package e2e
+	"github.com/operator-framework/operator-metering/test/deployframework"
 	"github.com/operator-framework/operator-metering/test/reportingframework"
+	"github.com/operator-framework/operator-metering/test/testhelpers"
 )
 
 var (
-	testReportingFramework *reportingframework.ReportingFramework
+	df *deployframework.DeployFramework
 
 	reportTestOutputDirectory string
+	testOutputDirectory       string
 	runAWSBillingTests        bool
 )
 
+/*
+TODO:
+- make sure that the metering and reporting operator images are overrided
+- update this PR so that the CI is passing w/o knowledge of local flags (https-api, use-kube-proxy...)
+  * these flags are more contingent on the meteringconfig spec
+  * https-api: spec.tls.enabled (default true)
+  * use-route: always true
+  * use-https: always true
+  * kube proxy: always false
+  * reporting API URL: setting this to empty
+
+Current Problems:
+- ioutil.TempDir has the behavior we want, but doesn't create parent directories
+- the reporting framework flags are hardcoded
+- need a way to be able to dump the actual test logs (and log to stdout)
+- need to support overriding the metering and reporting operator images
+- need to be able to create a base directory (and the parent if applicable) w/o permission denied error
+- the defer closure isn't respected when with require.NoError or t.Fatalf (t.FailNow), or require.FailNow
+- need to support decoding a METERING_CR_FILE for local testing?
+
+Questions:
+- should we be able to provide a test output directory, or should it always use a prefix + tempDir
+  * could change the `$TEST_OUTPUT_PATH` defined in openshift-release and remove the `/e2e` dir
+*/
+
 func init() {
-	reportTestOutputDirectory = os.Getenv("TEST_RESULT_REPORT_OUTPUT_DIRECTORY")
-	if reportTestOutputDirectory == "" {
-		log.Fatalf("$TEST_RESULT_REPORT_OUTPUT_DIRECTORY must be set")
+	testOutputDirectory = os.Getenv("TEST_OUTPUT_PATH")
+	runAWSBillingTests = os.Getenv("ENABLE_AWS_BILLING_TESTS") == "true"
+}
+
+func TestMain(m *testing.M) {
+	var err error
+
+	kubeConfigFlag := flag.String("kubeconfig", "", "kube config path, e.g. $HOME/.kube/config")
+	nsPrefix := flag.String("namespace-prefix", "", "The namespace prefix to install the metering resources.")
+	manifestDir := flag.String("deploy-manifests-dir", "../../manifests/deploy", "The absolute/relative path to the metering manifest directory.")
+	cleanupScriptPath := flag.String("cleanup-script-path", "../../hack/run-test-cleanup.sh", "The absolute/relative path to the testing cleanup hack script.")
+	testOutputPath := flag.String("test-output-path", "", "The absolute/relative path that you want to store test logs within.")
+	logLevel := flag.String("log-level", logrus.DebugLevel.String(), "The log level")
+	flag.Parse()
+
+	logger := testhelpers.SetupLogger(*logLevel)
+
+	if testOutputDirectory == "" {
+		if *testOutputPath == "" {
+			logger.Fatalf("You must specify the $TEST_OUTPUT_PATH or --test-output-path.")
```

Comment:
@chancez how do you want to handle specified directories (through `$TEST_OUTPUT_PATH/--test-output-path)? I think it makes more sense to use that path instead of making a directory with that path + a prefix (e.g. `ioutil.TempDir`).

We also need to either support making a temp directory when the flag/ENV var is unspecified or make it a required value, which doesn't seem ideal.

Reply:
We could do something like this, but seems awkward:
```go
	var loggingPath string
	var dirCreated bool

	switch {
	case testOutputDirectory != "":
		loggingPath = testOutputDirectory
	case *testOutputPathFlag != "":
		loggingPath = *testOutputPathFlag
	default:
		loggingPath, err = ioutil.TempDir(testOutputDirectory, *nsPrefix)
		if err != nil {
			logger.Fatalf("Failed to create the test output directory '%s': %v", testOutputDirectory, err)
		}
		dirCreated = true
	}

	if !dirCreated {
		err := os.MkdirAll(testOutputDirectory, 0777)
		if err != nil {
			logger.Fatalf("Failed to create the test output directory '%s': %v", testOutputDirectory, err)
		}
	}
```